dcpy/test/connectors/edm/test_recipes.py

```diff
@@ -47,11 +60,28 @@ def load_library_all(create_buckets):
     yield library_config
 
 
-def test_archive_dataset(create_buckets, create_temp_filesystem: Path):
+def _mock_template(name, version, path=None):
+    configure.read_template(name, version, RESOURCE_DIR)
+
+
+@pytest.fixture(scope="function")
+def load_ingest(create_buckets):
+    with mock.patch.object(read_template, "__defaults__", (None, None, RESOURCE_DIR)):
```

Comment:
Q about the generated mock object above. From my understanding:
* `read_template` is the method we are mocking
* `"__defaults__"` are `read_template` input arg names (ie `dataset`, `version`, and `template_dir`)
* `(None, None, RESOURCE_DIR)` are the values to call `read_template` with. 

My question is: how do we call with `dataset` = `None` if `dataset` is a required arg?

Reply:
This is patching the defaults to the function `read_template` - when it's called in a test, any supplied args at run time will overwrite these defaults. This function when called within `run` doesn't supply an argument for the `template_dir`, so it will use the default that I've defined here. The other two args are supplied, so it's fine that I didn't set a default here (or set a default of `None`)